homeassistant/components/template/config.py

```diff
@@ -252,3 +254,11 @@ async def async_validate_config(hass: HomeAssistant, config: ConfigType) -> Conf
     config[DOMAIN] = config_sections
 
     return config
+
+
+def _is_trigger_based_template_entity(config):
+    """Check if this is a trigger based template entity.
+
+    Takes into account backwards compatible definition.
+    """
+    return CONF_TRIGGERS in config or CONF_TRIGGER in config
```

Comment:
As I understand it, the configuration is rewritten and there can only be `triggers`, not `trigger` after schema validation. So why do the check both?